mp/importer/tests/test_batcher.py

```diff
@@ -0,0 +1,58 @@
+import os
+from unittest import TestCase
+import collections
+from collections import namedtuple
+from collections import Iterable
+from itertools import *
+
+def make_test_context(b_size, b_start, b_max):									# creates a "provisional" context 
+    context = namedtuple('Context', """
+        batch_size
+        start_batch
+        max_batches """)
+    context.batch_size = b_size
+    context.start_batch = b_start
+    context.max_batches = b_max
+    return context
+
+class TestBatcher(TestCase):
+
+    def one(self, source, batch_size=2, batch_start=0, max_batches=10):					# batch size cannot be 0
+        log = []
+        def end_batch():                
+            log.append('X')
+        from .. import batcher
+        context = make_test_context(batch_size, batch_start, max_batches)
+        batcher = batcher.run_in_batches(       
+                context,
+                source, end_batch)          
+        for letter in batcher:              
+            log.append(letter)
+        if log[-1] != 'X':
+            end_batch()
```

Comment:
This "fixes" a defect in the working code. It shouldn't either fix the code or change the tests.